### PR TITLE
add chip button component

### PR DESCRIPTION
**Pull request by radeknovis**

### CODE REVIEW

File: Sources/Web3ModalUI/Resources/Extensions/Font+extension.swift [hunk 1]
```diff
@@ -1,19 +1,118 @@
...
+        return content.font(.system(size: scaledSize, weight: weight))
+    }
+}
+
+extension View {
+    public func scaledFont(size: Double, weight: Font.Weight) -> some View {
+        return modifier(ScaledFont(size: size, weight: weight))
+    }
+    
+    public func font(_ font: W3MFont) -> some View {
```
Comment (flypaper0) [inline Sources/Web3ModalUI/Resources/Extensions/Font+extension.swift:78]: method name is `font` but it still make font Scaled. Everything fine here?

File: Sources/Web3ModalUI/Resources/Extensions/Font+extension.swift [hunk 2]
```diff
@@ -1,19 +1,118 @@
...
+        case .tiny500: return 12.0
+        case .tiny600: return 12.0
+        case .title500: return 24.0
+        case .title600: return 24.0
+        case .title700: return 24.0
+        }
+    }
+}
+
+struct ScaledFont: ViewModifier {
```
Comment (flypaper0) [inline Sources/Web3ModalUI/Resources/Extensions/Font+extension.swift:61]: it's scale font depends on screen size or how it's working?

**Review by flypaper0 — APPROVED** [review]

File: Sources/Web3ModalUI/Resources/Extensions/Font+extension.swift (hunk 2 above)
Comment (radeknovis) [reply Sources/Web3ModalUI/Resources/Extensions/Font+extension.swift:61]: All the fonts are "Scaled". The problem here is if you want a `.system`  font of a specific size and weight, it no longer reacts to DynamicType size automatically, and you need to do this hack.  

More context here https://www.hackingwithswift.com/quick-start/swiftui/how-to-use-dynamic-type-with-a-custom-font

in our case we don't have a custom but system. If we don't do this we would need to use https://developer.apple.com/documentation/swiftui/font/custom(_:size:relativeto:)

File: Sources/Web3ModalUI/Resources/Extensions/Font+extension.swift (hunk 1 above)
Comment (radeknovis) [reply Sources/Web3ModalUI/Resources/Extensions/Font+extension.swift:78]: yeah that is on purpose, so we can keep the code as is using e.g. `.font(.large500)`

**Pull request merged**